src/sys/unix/net.rs

```diff
@@ -20,14 +20,14 @@ pub(crate) fn new_socket(domain: libc::c_int, socket_type: libc::c_int) -> io::R
         target_os = "illumos",
         target_os = "linux",
         target_os = "netbsd",
-        target_os = "openbsd"
+        target_os = "openbsd",
     ))]
     let socket_type = socket_type | libc::SOCK_NONBLOCK | libc::SOCK_CLOEXEC;
 
     let socket = syscall!(socket(domain, socket_type, 0))?;
 
     // Mimick `libstd` and set `SO_NOSIGPIPE` on apple systems.
-    #[cfg(target_vendor = "apple")]
+    #[cfg(any(target_os = "ios", target_os = "macos"))]
```

Comment:
Nit, take it or leave it: rust-lang/lang-team#102 also mentions the existence of `target_family = "apple"`, which is set in addition to `target_family = "unix"` when building for both iOS and macOS. It might be a little bit simpler to use `target_family = "apple"` here instead?